responsive sidenav

sandikbarr opened the pull request:
implements responsive side nav #285 and responsive menu and controls #170 

~TODO: NEEDS STYLEGUIDE UPDATE TO INCLUDE ICONS~ ✅ done

sandikbarr commented on src/app/downloads/downloads.component.ts:
```diff
@@ -96,8 +96,9 @@ export class DownloadsComponent implements OnInit, OnDestroy {
               this.podcast = this.podcasts.find(p => p.seriesId === this.routerState.podcastSeriesId);
-              this.totalPages = this.podcast.doc.count('prx:stories') / EPISODE_PAGE_SIZE;
-              if (this.podcast.doc.count('prx:stories') % EPISODE_PAGE_SIZE > 0) {
+              const numEpisodes = this.podcast ? this.podcast.doc.count('prx:stories') : 0;
```
stumbled on this bug with a podcast route that my newly logged in user was not authorized for

sandikbarr commented on src/app/shared/menu/date/custom-date-range-dropdown.component.ts:
```diff
@@ -46,11 +46,15 @@ import { GoogleAnalyticsEventAction, RouteAdvancedRangeAction } from '../../../n
   userChoseRange: string;
   CUSTOM_DATE = 'custom-date';
   STANDARD_DATE = 'standard-date';
+  open = false;
```
noticed especially in mobile mode when there is more content to scroll that these dropdowns remained open but were out of place when the window scrolled

cavis commented on src/app/shared/menu/date/standard-date-range-dropdown.component.ts:
```diff
@@ -32,8 +25,11 @@ import * as dateUtil from '../../util/date';
   open = false;
+  @HostListener('window: scroll', [])
+  onWindowScroll() {
+    this.open = false;
```
Would it be good to also do this on the other dropdowns?

cavis replied:
Ah, i see it's on the canned interval dropdown.  But maybe on the custom one too?

sandikbarr replied:
I did actually have it on the custom date range dropdown but I took it out. I found that at my screen resolution, the window needed to be almost as tall as the viewport to see the Apply button. So if the user tried to scroll to it, the dropdown closes, and everyone is sad. I added these because it was gross to me that the dropdowns were fixed on scroll, but not being able to use the is kind of worse,

cavis commented on src/app/shared/nav/nav-menu.component.ts:
```diff
@@ -1,50 +1,52 @@
+      <div class="dropdown-content rollout">
+        <nav>
+          <li *ngFor="let t of types">
+            <button
```
In mobile view ... when i click/select on these buttons, the dropdown doesn't close.  Have to click outside of the dropdown to make it close.

sandikbarr replied:
will take a look!

cavis commented on src/app/shared/podcast-nav/podcast-nav-dropdown.component.ts:
```diff
@@ -32,7 +32,7 @@ export class PodcastNavDropdownComponent {
 
   onPodcastChange(val: PodcastModel) {
-    if (val && val.seriesId !== this.selectedPodcast.seriesId) {
+    if (val && (!this.selectedPodcast || val.seriesId !== this.selectedPodcast.seriesId)) {
```
+1.  i think i've hit this before, when masquerading as different users.

cavis commented on src/app/shared/menu/date/custom-date-range-dropdown.component.ts:
```diff
@@ -10,7 +10,10 @@ import { GoogleAnalyticsEventAction, RouteAdvancedRangeAction } from '../../../n
       <div class="dropdown-button">
-        <button class="btn-icon icon-calendar grey-dove" (click)="toggleOpen()" aria-label="Custom Date Range"></button>
+        <button (click)="toggleOpen()">
+          <metrics-date-range-summary [routerState]="routerState"></metrics-date-range-summary>
```
I'm noting this component has Hourly/Daily/Weekly/Monthly buttons in it.  Are those necessary, since the dropdown for those is right next to this button?

sandikbarr replied:
We didn't have any changes to the content of the desktop dropdowns in the designs, so I'm not sure if we meant to keep the quick interval selector and have the advanced all option selector on desktop, or what, but this is how I interpreted it.

cavis commented on src/app/downloads/downloads.component.ts:
```diff
@@ -16,15 +16,14 @@ import { isPodcastChanged, isBeginDateChanged, isEndDateChanged, isIntervalChang
+    <section *ngIf="loaded$ | async">
       <metrics-menu-bar></metrics-menu-bar>
       <metrics-downloads-chart></metrics-downloads-chart>
       <metrics-downloads-table [totalPages]="totalPages"></metrics-downloads-table>
```
I'm noting that, in mobile, the table + pager get a little funky.  (Horizontal scrolling, page-number-overflow, etc).  Will that be addressed in a different ticket/PR?

sandikbarr replied:
There is a separate set of tickets to update the table.

sandikbarr replied:
tables design is here https://github.com/PRX/metrics.prx.org/issues/279

sandikbarr commented:
On the subject of page height, I message around with vh units on the `article` element. We don't have a defined height on the footer, so we can't really `calc(100vh - XZYpx)` but maybe we could do something different there. Without including height for the footer, it looked especially odd to me on tall mobile screens to push the footer down below a big empty COMING SOON screen.

I think we can continue making improvements on things like button size and other thoughts that come up. The purpose of this PR is to add responsive support to the menu bar (now a less defined bar but those same date range and interval controls) and to the side navigation, but I'm sure the mobile friendly conversation definitely isn't over.

I also liked the always visible series image. If we bring it back, I hope we can talk about why it was circular instead of square since podcast cover images are designed to be square.

Other than that, hopefully I've addressed your comments @cavis. Thanks!

pkarman commented:
lgtm!
![screen shot 2018-06-26 at 9 52 17 am](https://user-images.githubusercontent.com/1205061/41920655-c9dd0c2c-7926-11e8-846a-492cd9138c6a.png)
